packages/components/src/Tabs/hooks/useArrowKeyNavigation.ts

```diff
@@ -0,0 +1,47 @@
+import { useCallback } from "react";
+
+interface UseArrowKeyNavigationProps {
+  elementsRef: React.RefObject<(HTMLElement | null)[]>;
+  onActivate: (index: number) => void;
+}
+
+export const useArrowKeyNavigation = ({
```

Comment:
I'm choosing to create and keep this hook within `Tabs` for a few reasons:

1. We talked about sharing arrow key navigation between `SegmentedControl` and `Tabs` since they are very similar however, `SegmentedControl` does not have any specific arrow key logic.  That is because arrow key navigation is implicitly set through the `SegmentedControl` radio button structure. The options are a type of radio and part of a group (name attribute) which inherently supports keyboard navigation. So since the component uses a `role="radiogroup"` and `name` attribute, the browser handles this behaviour automatically. There's nothing to share between these two components.

2. There is one other area (`packages/components/src/Chips/InternalChipDismissible/hooks/useInternalChipDismissible.ts`) where this type of logic is being used but to be honest, creating a more universal version of this hook for one other component (that I'd prefer not to touch right now 😅 ) is not something I want to increase scope on.  I'm happy to create a ticket to create a more flexible hook in the future if we think that is something our library should have. I have ideas on what that hoook can look like so can document it in the ticket

Reply:
Love this breakdown

Reply:
Yeah that sounds good - I can play around with that today and see what you and Chris think!